tls/s2n_ktls.h

```diff
@@ -42,7 +42,16 @@ typedef int (*s2n_setsockopt_fn)(int socket, int level, int option_name,
 S2N_RESULT s2n_ktls_set_setsockopt_cb(s2n_setsockopt_fn cb);
 
 bool s2n_ktls_is_supported_on_platform();
+S2N_RESULT s2n_ktls_get_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
 
 /* These functions will be part of the public API. */
 int s2n_connection_ktls_enable_send(struct s2n_connection *conn);
 int s2n_connection_ktls_enable_recv(struct s2n_connection *conn);
+
+/* Testing */
+/* msghdr is not `const` for sendmsg because the real implementation will need to modify
+ * ancillary data on msghdr prior to sending */
+typedef ssize_t (*s2n_ktls_sendmsg_fn)(struct s2n_connection *conn, struct msghdr *msg, uint8_t record_type);
+typedef ssize_t (*s2n_ktls_recvmsg_fn)(struct s2n_connection *conn, struct msghdr *msg, uint8_t *record_type);
```

Comment:
With https://github.com/aws/s2n-tls/pull/4109/files#r1274167851 in mind, it's also worth pointing out that these signatures don't match the send/recv callbacks. Most of our callbacks, including s2n_send_fn and s2n_recv_fn, take the context pointer directly. s2n_send_fn and s2n_recv_fn also take the fd instead of retrieving it from the connection.

On the one hand, this works as-is. On the other hand, we've run into issues / caused friction for future developers when we don't match existing patterns.